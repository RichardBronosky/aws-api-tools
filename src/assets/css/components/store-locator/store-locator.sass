@import '../../mixins'
@import '../../variables'

.store-locator-page
