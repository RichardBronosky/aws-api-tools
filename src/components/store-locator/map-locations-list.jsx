import React from 'react'
import classNames from 'classnames'
import RTGLink from '../../components/shared/link'
import { letters, getStandardTime, weekdays } from '../../lib/helpers/string-helper.js'
import '../../assets/css/components/store-locator/map-locations-list.sass'

const showDay = (hour, index, currentDay, tomorrowDay) => {
  let daytxt = false
  const day = hour.day ? hour.day : weekdays[hour.dayIndex]
  if (day == currentDay) daytxt = 'Today'
  if (day == tomorrowDay) daytxt = 'Tomorrow'
  if (daytxt) {
    return (
      <div className="hour" key={ index }>
        <span className="day">{ daytxt }</span>: { getStandardTime(hour.openTime) } - { getStandardTime(hour.closeTime) }
      </div>
    )
  }
}

export default ({ markers, onLocationClick, showHours, toggleShowHours, currentDay, tomorrowDay, lia }) => (
  <div className="locations">
    { markers.map((mark, index) => {
      let hours
      if (mark.hours && !mark.hours.regularHours) {
        hours = mark.hours
      } else if (mark.hours && mark.hours.regularHours) {
        hours = mark.hours.regularHours
      }
      const phone = mark.phone || mark.phoneNumber
      return (
        <div className="locationRow" onClick={ () => onLocationClick(index, mark) } key={ index }>
          <div className="imgCol">
            <img src={ `http://maps.google.com/mapfiles/marker${ letters[index] }.png` } />
          </div>
          <div className="addrCol">
            <span className="storeName">
              { !lia ? (
                mark.name
              ) : (
                <RTGLink
                  data={ {
                    url: window && `${ window.location.pathname }?storeCode=${ mark.storeNumber }`,
                    title: `Change Location to ${ mark.name } Store`,
                    category: 'product-lia',
                    action: 'change location click',
                    label: mark.name,
                  } }
                >
                  { mark.name }
                </RTGLink>
              ) }
            </span>
            <br />
            <div className="phone">
              <RTGLink
                data={ {
                  url: `tel:${ phone.replace('/', '-') }`,
                  title: `Call ${ mark.name } Store`,
                  category: 'store-locator',
                  action: 'call store click',
                  label: mark.name,
                } }
              >
                { phone.replace('/', '-') }
              </RTGLink>
            </div>
            <div className="address">
              <RTGLink
                data={ {
                  url: `https://www.google.com/maps/dir/?api=1&origin=Current+Location&destination=${ mark.lat },${ mark.lng }`,
                  title: `Directions to ${ mark.name }`,
                  category: 'store locator',
                  action: 'Directions store map click',
                  label: 'Directions to store',
                } }
                target="_blank"
                className="store-address-link"
              >
                { mark.title }
                <br />
                { mark.city }, { mark.state } { mark.zip }
              </RTGLink>
              <br />
              { mark.distance && <>{ mark.distance } miles</> }
            </div>
            <div className="hours">
              { mark.specialClosings && (
                <span
                  className="special-closings"
                  dangerouslySetInnerHTML={ { __html: mark.specialClosings.markdown.childMarkdownRemark.html } }
                />
              ) }
              <div className="heading">Store Hours</div>
              { hours &&
                hours.map((hour, i) => {
                  if (!showHours[index]) {
                    return showDay(hour, i, currentDay, tomorrowDay)
                  } else {
                    return (
                      <div
                        className={ classNames('hour', {
                          hidden: i > 0 && !showHours[index],
                        }) }
                        key={ i }
                      >
                        <span className="day">{ hour.day ? hour.day : weekdays[hour.dayIndex] }</span>:{ ' ' }
                        { hour.openTime && hour.closeTime
                          ? `${ getStandardTime(hour.openTime) } - ${ getStandardTime(hour.closeTime) }`
                          : 'Closed' }
                      </div>
                    )
                  }
                }) }
            </div>
            <div>
              <button className="showAllBtn" value="Show All" onClick={ () => toggleShowHours(index) }>
                { !showHours[index] ? 'Show All' : 'Show Less' }
              </button>
            </div>
          </div>
        </div>
      )
    }) }
  </div>
)
