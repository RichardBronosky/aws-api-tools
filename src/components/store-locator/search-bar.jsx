import React from 'react'
import classNames from 'classnames'
import '../../assets/css/components/store-locator/search-bar.sass'
import { keyboardBlur } from '../../lib/helpers/input-helper'

const onEnterPress = (e, changeZip) => {
  if (e.keyCode == 13 && e.shiftKey == false) {
    changeZip()
    keyboardBlur(e)
  }
}

export default ({ title, invalidZip, zipInput, updateInputZipValue, changeZip }) => {
  return (
    <div className="searchbar">
      <span className="seeStore">{ title }</span>
      <label
        className={ classNames({
          invalid: invalidZip,
        }) }
        htmlFor="location"
      >
        { invalidZip && <p className="invalid-message">*Please enter a valid ZIP code</p> }
        <p className="label-text">Enter ZIP code</p>
        <input
          type="text"
          className="zip-input"
          name="location"
          id="location"
          placeholder="ZIP code"
          aria-label="Enter zip code"
          value={ zipInput }
          onChange={ evt => updateInputZipValue(evt) }
          onKeyDown={ e => onEnterPress(e, changeZip) }
        />
        <button
          value="Change zip code"
          name="change_zip"
          className="submit-btn"
          id="submit-btn"
          aria-label="Change zip code"
          gtm-category="see in store"
          gtm-action="change zip code"
          onClick={ changeZip }
        >
          Submit
        </button>
      </label>
    </div>
  )
}
