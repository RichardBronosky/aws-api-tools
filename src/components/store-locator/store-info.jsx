import React from 'react'
import { Provider } from 'react-redux'
import { store } from '../../redux/store'
import RTGLink from '../shared/link'
import { weekdays, getStandardTime } from '../../lib/helpers/string-helper'
import '../../assets/css/components/store-locator/store-info.sass'

export default ({ selectedMarker }) => {
  let hours
  if (selectedMarker.hours && !selectedMarker.hours.regularHours) {
    hours = selectedMarker.hours
  } else if (selectedMarker.hours && selectedMarker.hours.regularHours) {
    hours = selectedMarker.hours.regularHours
  }
  const currentDay = weekdays[new Date().getDay()]
  return (
    <Provider store={ store }>
      <div className="store-info">
        <h2 className="store-name">{ selectedMarker.name }</h2>
        <div className="storeAddress">
          <RTGLink
            data={ {
              url: `https://www.google.com/maps/dir/?api=1&origin=Current+Location&destination=${ selectedMarker.lat },${ selectedMarker.lng }`,
              title: `Directions to ${ selectedMarker.name }`,
              category: 'store locator',
              action: 'Directions store map click',
              label: 'Directions to store',
            } }
            target="_blank"
            className="store-address-link"
          >
            { selectedMarker.title }
            <br />
            { selectedMarker.city }, { selectedMarker.state } { selectedMarker.zip }
          </RTGLink>
          <br />
          { (selectedMarker.phone || selectedMarker.phoneNumber) && (
            <RTGLink
              data={ {
                url: `tel:${ (selectedMarker.phone ? selectedMarker.phone : selectedMarker.phoneNumber).replace(
                  '/',
                  '-'
                ) }`,
                title: `Call ${ selectedMarker.name }`,
                category: 'store locator',
                action: 'Call store map click',
                label: 'Call store',
              } }
            >
              { (selectedMarker.phone ? selectedMarker.phone : selectedMarker.phoneNumber).replace('/', '-') }
            </RTGLink>
          ) }
          { hours && (
            <div className="today-hours">
              { hours &&
                hours.map((hour, i) => {
                  const day = hour.day ? hour.day : weekdays[hour.dayIndex]
                  if (day === currentDay) {
                    return (
                      <div className="hour" key={ i }>
                        <span className="day">Today</span>: { getStandardTime(hour.openTime) } -{ ' ' }
                        { getStandardTime(hour.closeTime) }
                      </div>
                    )
                  }
                }) }
            </div>
          ) }
        </div>
      </div>
    </Provider>
  )
}
