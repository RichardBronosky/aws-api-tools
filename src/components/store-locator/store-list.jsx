import React from 'react'
import RTGLink from '../shared/link'
import { expandState } from '../../lib/helpers/geo-location.js'
import { slugify } from '../../lib/helpers/string-helper'
import '../../assets/css/components/store-locator/store-list.sass'

export default class StoreList extends React.Component {
  render() {
    const { storeLocations: stores, states, abbrevStates } = this.props
    return (
      <div className="store-list">
        <h2>Store Listings by State</h2>
        <div className="state-list-container">
          { abbrevStates.map((state, index) => (
            <div key={ index } className="state-list">
              <h3>
                <RTGLink
                  data={ {
                    slug: `/stores/${ slugify(states[index]) }`,
                    category: 'store',
                    action: 'click',
                    label: `All stores in ${ states[index] }`,
                  } }
                >
                  { states[index] }
                </RTGLink>
              </h3>
              { stores
                .filter(loc => loc.state === state && loc.storeTypeId !== 'W')
                .map((loc, index) => {
                  let type
                  if (loc.storeTypeId === 'A') {
                    type = 'Showroom'
                  } else if (loc.storeTypeId === 'B') {
                    type = 'EXPRESS'
                  } else if (loc.storeTypeId === 'C') {
                    type = 'Outlet'
                  } else if (loc.storeTypeId === 'K') {
                    type = 'Kids'
                  } else if (loc.storeTypeId === 'P') {
                    type = 'Outdoor'
                  }
                  if (loc.storeName) {
                    return (
                      <div className="link-container" key={ loc.id }>
                        <RTGLink
                          data={ {
                            url: `/stores/${ slugify(expandState(state)).toLowerCase() }/${ slugify(
                              loc.city.toLowerCase()
                            ) }-${ slugify(loc.storeName.toLowerCase()) }-${
                              type ? slugify(type.toLowerCase()) : slugify(loc.storeType.toLowerCase())
                            }-${ loc.storeNumber }`,
                            text: `${ loc.city } - ${ loc.storeName } ${ type || loc.storeType }`,
                            category: 'store',
                            action: 'click',
                            label: `${ loc.city } - ${ loc.storeName } ${ type || loc.storeType }`,
                            value: index,
                          } }
                        />
                      </div>
                    )
                  } else {
                    return (
                      <div className="link-container" key={ loc.id }>
                        <RTGLink
                          data={ {
                            url: `/stores/${ slugify(expandState(state)).toLowerCase() }/${ slugify(
                              loc.city.toLowerCase()
                            ) }-${ type ? slugify(type.toLowerCase()) : slugify(loc.storeType.toLowerCase()) }-${
                              loc.storeNumber
                            }`,
                            text: `${ loc.city } ${ type || loc.storeType }`,
                            category: 'store',
                            action: 'click',
                            label: `${ loc.city } ${ type || loc.storeType }`,
                            value: index,
                          } }
                        />
                      </div>
                    )
                  }
                }) }
            </div>
          )) }
        </div>
      </div>
    )
  }
}
