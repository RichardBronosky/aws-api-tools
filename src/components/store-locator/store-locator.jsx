import React from 'react'
import StoreMap from './store-map'
import StoreList from './store-list'
import '../../assets/css/components/store-locator/store-locator.sass'
import { setupAnalytics } from '../../lib/helpers/google-tag-manager.js'

export default class StoreLocator extends React.Component {
  componentDidMount() {
    setupAnalytics({ pageData: { type: 'store', title: 'Store Locator', path: '/stores' } })
  }

  render() {
    const { data } = this.props
    return (
      <div className="store-locator-page">
        <section className="seo-page-heading full-width hide show">
          <div className="grid-container ">
            <div className="grid-x grid-margin-x">
              <div className="cell small-12">
                <h1>Rooms To Go Store Locator</h1>
              </div>
            </div>
          </div>
        </section>
        <StoreMap storeLocations={ data.stores } />
        <StoreList storeLocations={ data.stores } states={ data.states } abbrevStates={ data.abbrevStates } />
      </div>
    )
  }
}
