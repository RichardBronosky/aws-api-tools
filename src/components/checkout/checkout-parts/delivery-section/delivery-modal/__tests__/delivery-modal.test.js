import React from 'react'
import renderer from 'react-test-renderer'
import ReactModal from 'react-modal'
import { Provider } from 'react-redux'
import { store } from '../../../../../../redux/store'
import DeliveryModal from '../delivery-modal'
import * as storeLocatorHelper from '../../../../../../lib/helpers/store-locator'
import { order1 } from '../../../../../../lib/mocks/checkoutDataMocks'
import { mockStore1 } from '../../../../../../lib/mocks/storeDataMocks'

ReactModal.setAppElement('*')

describe('DeliveryModal', () => {
  const testProps = {
    order: order1,
    modalOpen: true,
    closeModal: jest.fn(),
    setSelectedPickupDate: jest.fn(),
    setSelectedDeliveryDate: jest.fn(),
    loading: false,
    deliveryModalInfo: {
      delivery: true,
      pickup: false,
    },
    setDeliveryModalInfo: jest.fn(),
    error: '',
    deliveryCalendar: [
      {
        date: '2019-07-27',
        readable: 'July 27',
        dayOfWeek: 'Sat',
        isPickup: false,
        isStandardDelivery: true,
        isExpressDelivery: false,
      },
    ],
    selectedPickupDate: {
      date: '2019-07-27',
      readable: 'July 27',
    },
    selectedDeliveryDate: {
      date: '2019-07-27',
      readable: 'July 27',
    },
    updateDeliveryDate: jest.fn(),
  }

  it('renders correctly when open and delivery date selected', () => {
    storeLocatorHelper.getStore = jest.fn(() => mockStore1)
    const tree = renderer
      .create(
        <Provider store={ store }>
          <DeliveryModal { ...testProps } />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when open and express delivery date selected', () => {
    storeLocatorHelper.getStore = jest.fn(() => mockStore1)
    const newTestProps = {
      ...testProps,
      deliveryCalendar: [
        {
          date: '2019-07-27',
          readable: 'July 27',
          dayOfWeek: 'Sat',
          isPickup: false,
          isStandardDelivery: true,
          isExpressDelivery: true,
        },
      ],
    }
    const tree = renderer
      .create(
        <Provider store={ store }>
          <DeliveryModal { ...newTestProps } />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when open and pickup date selected', () => {
    storeLocatorHelper.getStore = jest.fn(() => mockStore1)
    const newTestProps = {
      ...testProps,
      deliveryModalInfo: {
        delivery: false,
        pickup: true,
      },
      deliveryCalendar: [
        {
          date: '2019-07-27',
          readable: 'July 27',
          dayOfWeek: 'Sat',
          isPickup: true,
          isStandardDelivery: true,
          isExpressDelivery: false,
        },
      ],
    }
    const tree = renderer
      .create(
        <Provider store={ store }>
          <DeliveryModal { ...newTestProps } />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when not open', () => {
    storeLocatorHelper.getStore = jest.fn(() => mockStore1)
    const newTestProps = {
      ...testProps,
      modalOpen: false,
    }
    const tree = renderer.create(<DeliveryModal { ...newTestProps } />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
