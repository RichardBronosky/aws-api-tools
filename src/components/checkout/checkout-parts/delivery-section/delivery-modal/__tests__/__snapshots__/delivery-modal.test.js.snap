// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryModal renders correctly when not open 1`] = `
<div>
  <div
    id="content"
  />
  <div
    data-target-tag-name="DIV"
    id="target"
  />
</div>
`;

exports[`DeliveryModal renders correctly when open and delivery date selected 1`] = `
<div>
  <div
    id="content"
  >
    <div
      className="ReactModal__Overlay ReactModal__Overlay--after-open modal-overlay"
      onClick={[Function]}
      onMouseDown={[Function]}
      style={Object {}}
    >
      <div
        aria-label="Add To Cart Modal"
        className="ReactModal__Content ReactModal__Content--after-open modal"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        role="dialog"
        style={Object {}}
        tabIndex="-1"
      >
        <div
          className="modal-content"
        >
          <div
            className="grid-x"
          >
            <div
              className="content-container"
            >
              <div
                className="cell small-12 title"
              >
                <h2>
                  Change Your Delivery
                </h2>
              </div>
              <div
                className="cell small-12"
              >
                <div
                  className="grid-x grid-margin-x grid-margin-y"
                >
                  <div
                    className="horizontal-center cell small-12 medium-4 medium-offset-2"
                  />
                </div>
              </div>
              <div
                className="cell radio-container small-12"
              >
                <input
                  aria-invalid={null}
                  className="text-me"
                  id="deliver"
                  name="deliver"
                  onChange={[Function]}
                  onClick={[Function]}
                  placeholder=""
                  type="radio"
                  value={true}
                />
                <span>
                  <span
                    className="radio-label"
                  />
                </span>
                <img
                  alt=""
                  aria-hidden="true"
                  className="icon shipping"
                  role="presentation"
                  src="data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%22100%25%22%20height%3D%22100%25%22%20viewBox%3D%220%200%2096.4%2049.1%22%3E%3Ctitle%3Eicon-shipping%3C%2Ftitle%3E%3Cg%20id%3D%22Layer_2%22%20data-name%3D%22Layer%202%22%3E%3Cg%20id%3D%22Layer_1-2%22%20data-name%3D%22Layer%201%22%3E%3Cpath%20d%3D%22M26.3%2C43.2H33a1.27%2C1.27%2C0%2C0%2C0%2C.6-.1%2C7%2C7%2C0%2C0%2C0%2C13.8.1h28a7%2C7%2C0%2C0%2C0%2C13.8%2C0h5.5a1.75%2C1.75%2C0%2C0%2C0%2C1.7-1.7V22.8c0-4.8-1.3-8.7-3.9-11.5a12.75%2C12.75%2C0%2C0%2C0-9.5-4H73V1.7A1.75%2C1.75%2C0%2C0%2C0%2C71.3%2C0h-45a1.75%2C1.75%2C0%2C0%2C0-1.7%2C1.7V41.5A1.75%2C1.75%2C0%2C0%2C0%2C26.3%2C43.2Zm56%2C2.4A3.6%2C3.6%2C0%2C1%2C1%2C85.9%2C42%2C3.59%2C3.59%2C0%2C0%2C1%2C82.3%2C45.6ZM73%2C10.7H83c6.2%2C0%2C10%2C4.6%2C10%2C12v17H88.9A7.14%2C7.14%2C0%2C0%2C0%2C82.3%2C35a7%2C7%2C0%2C0%2C0-6.6%2C4.7H73.1v-29ZM40.6%2C45.6A3.6%2C3.6%2C0%2C1%2C1%2C44.2%2C42%2C3.72%2C3.72%2C0%2C0%2C1%2C40.6%2C45.6ZM28%2C3.4H69.5V39.7H47.8a1.27%2C1.27%2C0%2C0%2C0-.6.1A7%2C7%2C0%2C0%2C0%2C40.5%2C35a7.12%2C7.12%2C0%2C0%2C0-6.7%2C4.9%2C1.88%2C1.88%2C0%2C0%2C0-.8-.2H28ZM0%2C11.8a1.75%2C1.75%2C0%2C0%2C1%2C1.7-1.7H18.1a1.7%2C1.7%2C0%2C0%2C1%2C0%2C3.4H1.7A1.75%2C1.75%2C0%2C0%2C1%2C0%2C11.8ZM18.1%2C23.3H4.3a1.7%2C1.7%2C0%2C0%2C1%2C0-3.4H18.1a1.7%2C1.7%2C0%2C0%2C1%2C0%2C3.4Zm-13.2%2C8a1.75%2C1.75%2C0%2C0%2C1%2C1.7-1.7H18A1.7%2C1.7%2C0%2C1%2C1%2C18%2C33H6.6A1.63%2C1.63%2C0%2C0%2C1%2C4.9%2C31.3Z%22%2F%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
                />
                <div
                  className="delivery-selection"
                >
                  <div
                    className="prof-delivery"
                  >
                    Professional delivery and set-up to any room
                  </div>
                  <p
                    className="bold"
                  >
                    $
                  </p>
                  <p
                    className="delivered-on"
                  >
                    Delivered on:
                  </p>
                  <p
                    className="bold margin-left"
                  >
                    July 27
                  </p>
                  <p
                    className="bold"
                  >
                    4 hour delivery window
                  </p>
                  <p
                    className="automated-message"
                  >
                    Our automated system will call 2 days prior to notify you of the estimated delivery window.
                  </p>
                </div>
              </div>
            </div>
            <div
              className="update-container cell small-12"
            >
              
              <button
                aria-label="Update Delivery"
                className="update-delivery"
                onClick={[Function]}
                tabIndex="0"
                value="Update Delivery"
              >
                update delivery
              </button>
            </div>
            <button
              aria-label="Close"
              className="close-modal"
              onClick={[MockFunction]}
              tabIndex="0"
              value="Close"
            >
              <img
                alt="close icon"
                className="icon close"
                src="data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cpath fill='%23FFFFFF' d='M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z' /%3E%3C/svg%3E"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-target-tag-name="DIV"
    id="target"
  />
</div>
`;

exports[`DeliveryModal renders correctly when open and express delivery date selected 1`] = `
<div>
  <div
    id="content"
  >
    <div
      className="ReactModal__Overlay ReactModal__Overlay--after-open modal-overlay"
      onClick={[Function]}
      onMouseDown={[Function]}
      style={Object {}}
    >
      <div
        aria-label="Add To Cart Modal"
        className="ReactModal__Content ReactModal__Content--after-open modal"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        role="dialog"
        style={Object {}}
        tabIndex="-1"
      >
        <div
          className="modal-content"
        >
          <div
            className="grid-x"
          >
            <div
              className="content-container"
            >
              <div
                className="cell small-12 title"
              >
                <h2>
                  Change Your Delivery
                </h2>
              </div>
              <div
                className="cell small-12"
              >
                <div
                  className="grid-x grid-margin-x grid-margin-y"
                >
                  <div
                    className="horizontal-center cell small-12 medium-4 medium-offset-2"
                  />
                </div>
              </div>
              <div
                className="cell radio-container small-12"
              >
                <input
                  aria-invalid={null}
                  className="text-me"
                  id="deliver"
                  name="deliver"
                  onChange={[Function]}
                  onClick={[Function]}
                  placeholder=""
                  type="radio"
                  value={true}
                />
                <span>
                  <span
                    className="radio-label"
                  />
                </span>
                <img
                  alt=""
                  aria-hidden="true"
                  className="icon shipping"
                  role="presentation"
                  src="data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%22100%25%22%20height%3D%22100%25%22%20viewBox%3D%220%200%2096.4%2049.1%22%3E%3Ctitle%3Eicon-shipping%3C%2Ftitle%3E%3Cg%20id%3D%22Layer_2%22%20data-name%3D%22Layer%202%22%3E%3Cg%20id%3D%22Layer_1-2%22%20data-name%3D%22Layer%201%22%3E%3Cpath%20d%3D%22M26.3%2C43.2H33a1.27%2C1.27%2C0%2C0%2C0%2C.6-.1%2C7%2C7%2C0%2C0%2C0%2C13.8.1h28a7%2C7%2C0%2C0%2C0%2C13.8%2C0h5.5a1.75%2C1.75%2C0%2C0%2C0%2C1.7-1.7V22.8c0-4.8-1.3-8.7-3.9-11.5a12.75%2C12.75%2C0%2C0%2C0-9.5-4H73V1.7A1.75%2C1.75%2C0%2C0%2C0%2C71.3%2C0h-45a1.75%2C1.75%2C0%2C0%2C0-1.7%2C1.7V41.5A1.75%2C1.75%2C0%2C0%2C0%2C26.3%2C43.2Zm56%2C2.4A3.6%2C3.6%2C0%2C1%2C1%2C85.9%2C42%2C3.59%2C3.59%2C0%2C0%2C1%2C82.3%2C45.6ZM73%2C10.7H83c6.2%2C0%2C10%2C4.6%2C10%2C12v17H88.9A7.14%2C7.14%2C0%2C0%2C0%2C82.3%2C35a7%2C7%2C0%2C0%2C0-6.6%2C4.7H73.1v-29ZM40.6%2C45.6A3.6%2C3.6%2C0%2C1%2C1%2C44.2%2C42%2C3.72%2C3.72%2C0%2C0%2C1%2C40.6%2C45.6ZM28%2C3.4H69.5V39.7H47.8a1.27%2C1.27%2C0%2C0%2C0-.6.1A7%2C7%2C0%2C0%2C0%2C40.5%2C35a7.12%2C7.12%2C0%2C0%2C0-6.7%2C4.9%2C1.88%2C1.88%2C0%2C0%2C0-.8-.2H28ZM0%2C11.8a1.75%2C1.75%2C0%2C0%2C1%2C1.7-1.7H18.1a1.7%2C1.7%2C0%2C0%2C1%2C0%2C3.4H1.7A1.75%2C1.75%2C0%2C0%2C1%2C0%2C11.8ZM18.1%2C23.3H4.3a1.7%2C1.7%2C0%2C0%2C1%2C0-3.4H18.1a1.7%2C1.7%2C0%2C0%2C1%2C0%2C3.4Zm-13.2%2C8a1.75%2C1.75%2C0%2C0%2C1%2C1.7-1.7H18A1.7%2C1.7%2C0%2C1%2C1%2C18%2C33H6.6A1.63%2C1.63%2C0%2C0%2C1%2C4.9%2C31.3Z%22%2F%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
                />
                <div
                  className="delivery-selection"
                >
                  <div
                    className="prof-delivery"
                  >
                    Professional delivery and set-up to any room
                  </div>
                  <p
                    className="bold"
                  >
                    $
                  </p>
                  <p
                    className="delivered-on"
                  >
                    Delivered on:
                  </p>
                  <p
                    className="bold margin-left"
                  >
                    July 27
                  </p>
                  <p
                    className="bold"
                  >
                    4 hour delivery window
                  </p>
                </div>
              </div>
            </div>
            <div
              className="update-container cell small-12"
            >
              
              <button
                aria-label="Update Delivery"
                className="update-delivery"
                onClick={[Function]}
                tabIndex="0"
                value="Update Delivery"
              >
                update delivery
              </button>
            </div>
            <button
              aria-label="Close"
              className="close-modal"
              onClick={[MockFunction]}
              tabIndex="0"
              value="Close"
            >
              <img
                alt="close icon"
                className="icon close"
                src="data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cpath fill='%23FFFFFF' d='M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z' /%3E%3C/svg%3E"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-target-tag-name="DIV"
    id="target"
  />
</div>
`;

exports[`DeliveryModal renders correctly when open and pickup date selected 1`] = `
<div>
  <div
    id="content"
  >
    <div
      className="ReactModal__Overlay ReactModal__Overlay--after-open modal-overlay"
      onClick={[Function]}
      onMouseDown={[Function]}
      style={Object {}}
    >
      <div
        aria-label="Add To Cart Modal"
        className="ReactModal__Content ReactModal__Content--after-open modal"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        role="dialog"
        style={Object {}}
        tabIndex="-1"
      >
        <div
          className="modal-content"
        >
          <div
            className="grid-x"
          >
            <div
              className="content-container"
            >
              <div
                className="cell small-12 title"
              >
                <h2>
                  Change Your Delivery
                </h2>
              </div>
              <div
                className="cell small-12"
              >
                <div
                  className="grid-x grid-margin-x grid-margin-y"
                >
                  <div
                    className="horizontal-center cell small-12 medium-4 medium-offset-2"
                  />
                </div>
              </div>
              <div
                className="cell radio-container small-12"
              >
                <input
                  aria-invalid={null}
                  className="text-me"
                  id="deliver"
                  name="deliver"
                  onChange={[Function]}
                  onClick={[Function]}
                  placeholder=""
                  type="radio"
                  value={true}
                />
                <span>
                  <span
                    className="radio-label"
                  />
                </span>
                <img
                  alt=""
                  aria-hidden="true"
                  className="icon shipping"
                  role="presentation"
                  src="data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%22100%25%22%20height%3D%22100%25%22%20viewBox%3D%220%200%2096.4%2049.1%22%3E%3Ctitle%3Eicon-shipping%3C%2Ftitle%3E%3Cg%20id%3D%22Layer_2%22%20data-name%3D%22Layer%202%22%3E%3Cg%20id%3D%22Layer_1-2%22%20data-name%3D%22Layer%201%22%3E%3Cpath%20d%3D%22M26.3%2C43.2H33a1.27%2C1.27%2C0%2C0%2C0%2C.6-.1%2C7%2C7%2C0%2C0%2C0%2C13.8.1h28a7%2C7%2C0%2C0%2C0%2C13.8%2C0h5.5a1.75%2C1.75%2C0%2C0%2C0%2C1.7-1.7V22.8c0-4.8-1.3-8.7-3.9-11.5a12.75%2C12.75%2C0%2C0%2C0-9.5-4H73V1.7A1.75%2C1.75%2C0%2C0%2C0%2C71.3%2C0h-45a1.75%2C1.75%2C0%2C0%2C0-1.7%2C1.7V41.5A1.75%2C1.75%2C0%2C0%2C0%2C26.3%2C43.2Zm56%2C2.4A3.6%2C3.6%2C0%2C1%2C1%2C85.9%2C42%2C3.59%2C3.59%2C0%2C0%2C1%2C82.3%2C45.6ZM73%2C10.7H83c6.2%2C0%2C10%2C4.6%2C10%2C12v17H88.9A7.14%2C7.14%2C0%2C0%2C0%2C82.3%2C35a7%2C7%2C0%2C0%2C0-6.6%2C4.7H73.1v-29ZM40.6%2C45.6A3.6%2C3.6%2C0%2C1%2C1%2C44.2%2C42%2C3.72%2C3.72%2C0%2C0%2C1%2C40.6%2C45.6ZM28%2C3.4H69.5V39.7H47.8a1.27%2C1.27%2C0%2C0%2C0-.6.1A7%2C7%2C0%2C0%2C0%2C40.5%2C35a7.12%2C7.12%2C0%2C0%2C0-6.7%2C4.9%2C1.88%2C1.88%2C0%2C0%2C0-.8-.2H28ZM0%2C11.8a1.75%2C1.75%2C0%2C0%2C1%2C1.7-1.7H18.1a1.7%2C1.7%2C0%2C0%2C1%2C0%2C3.4H1.7A1.75%2C1.75%2C0%2C0%2C1%2C0%2C11.8ZM18.1%2C23.3H4.3a1.7%2C1.7%2C0%2C0%2C1%2C0-3.4H18.1a1.7%2C1.7%2C0%2C0%2C1%2C0%2C3.4Zm-13.2%2C8a1.75%2C1.75%2C0%2C0%2C1%2C1.7-1.7H18A1.7%2C1.7%2C0%2C1%2C1%2C18%2C33H6.6A1.63%2C1.63%2C0%2C0%2C1%2C4.9%2C31.3Z%22%2F%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
                />
                <div
                  className="delivery-selection"
                >
                  <div
                    className="prof-delivery"
                  >
                    Professional delivery and set-up to any room
                  </div>
                  <p
                    className="bold"
                  >
                    $
                  </p>
                  <p
                    className="delivered-on"
                  >
                    Delivered on:
                  </p>
                  <p
                    className="bold margin-left"
                  >
                    July 27
                  </p>
                  <p
                    className="bold"
                  >
                    4 hour delivery window
                  </p>
                  <p
                    className="automated-message"
                  >
                    Our automated system will call 2 days prior to notify you of the estimated delivery window.
                  </p>
                </div>
              </div>
            </div>
            <div
              className="update-container cell small-12"
            >
              
              <button
                aria-label="Update Delivery"
                className="update-delivery"
                onClick={[Function]}
                tabIndex="0"
                value="Update Delivery"
              >
                update delivery
              </button>
            </div>
            <button
              aria-label="Close"
              className="close-modal"
              onClick={[MockFunction]}
              tabIndex="0"
              value="Close"
            >
              <img
                alt="close icon"
                className="icon close"
                src="data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cpath fill='%23FFFFFF' d='M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z' /%3E%3C/svg%3E"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-target-tag-name="DIV"
    id="target"
  />
</div>
`;
